File: lambda_functions/addTask/addTask.js
const AWS = require('aws-sdk');
const dynamoDb = new AWS.DynamoDB.DocumentClient();

exports.handler = async (event) => {
  try {
    // Ensure that the event body is correctly parsed
    let data;
    if (event.body) {
      data = JSON.parse(event.body);
    } else {
      throw new Error('No data provided');
    }

    // Extract fields from the parsed data
    const { id, title, description, date, priority, completed } = data;

    // Validate required fields
    if (!id || !title || !description || !date || !priority || completed === undefined) {
      throw new Error('Missing required fields');
    }

    // Set up parameters for DynamoDB
    const params = {
      TableName: 'Tasks',
      Item: {
        id,
        title,
        description,
        date,
        priority,
        completed
      }
    };

    // Put item into DynamoDB
    await dynamoDb.put(params).promise();

    return {
      statusCode: 200,
      body: JSON.stringify({ message: 'Task added successfully!' })
    };

  } catch (error) {
    return {
      statusCode: 400,
      body: JSON.stringify({ error: error.message })
    };
  }
};

File: lambda_functions/addTask/addTask.py
import json
import boto3

dynamo_db = boto3.client('dynamodb')

def lambda_handler(event, context):
    try:
        # Ensure that the event body is correctly parsed
        data = json.loads(event.get('body', '{}'))
        
        # Extract fields from the parsed data
        id = data.get('id')
        title = data.get('title')
        description = data.get('description')
        date = data.get('date')
        priority = data.get('priority')
        completed = data.get('completed')
        
        # Validate required fields
        if not all([id, title, description, date, priority, completed is not None]):
            raise ValueError('Missing required fields')

        # Set up parameters for DynamoDB
        params = {
            'TableName': 'Tasks',
            'Item': {
                'id': {'S': id},
                'title': {'S': title},
                'description': {'S': description},
                'date': {'S': date},
                'priority': {'S': priority},
                'completed': {'BOOL': completed}
            }
        }
        
        # Put item into DynamoDB
        dynamo_db.put_item(**params)
        
        return {
            'statusCode': 200,
            'headers': {
                'Access-Control-Allow-Origin': '*',
                'Access-Control-Allow-Methods': 'GET, POST, OPTIONS',
                'Access-Control-Allow-Headers': 'Content-Type, Authorization'
            },
            'body': json.dumps({'message': 'Task added successfully!'})
        }
    
    except Exception as e:
        return {
            'statusCode': 400,
            'headers': {
                'Access-Control-Allow-Origin': '*',
                'Access-Control-Allow-Methods': 'GET, POST, OPTIONS',
                'Access-Control-Allow-Headers': 'Content-Type, Authorization'
            },
            'body': json.dumps({'error': str(e)})
        }
